### ISSUE TITLE
Support `Back` button/`KeyCode` on Android

### ISSUE THREAD

**Issue by MarijnS95** [rust-windowing/winit#2304]
Started playing about with these codes, and it seems you haven't implemented a mapping for `KeyCode::Back` which would be needed to hanlde a graceful app exit (or going to the previous page) when the user presses it. The implementation for `VirtualKeyCode::Back` reads:

https://github.com/rust-windowing/winit/blob/f04fa5d54f4ec10cdb6d084deeb79d3e6d27ae67/src/event.rs#L962-L964

So I don't think we should add a new `Back` `VirtualKeyCode` and rename existing uses to `Backspace`, as that's really hard to notice for end-users upgrading their `winit`. Perhaps we can piggyback `NavigateBackward` but android already supports that as `KeyCode::NavigatePrevious`:

```rust
Keycode::Back => Some(VirtualKeyCode::NavigateBackward),
```

How do you think we should go forward?

_Originally posted by @MarijnS95 in https://github.com/rust-windowing/winit/pull/2226#discussion_r882830110_

**Comment by MarijnS95** [comment]
@kchibisov I don't see `Keycode::Back` being handled anywhere in Android's `platform_impl`, so I guess not?

**Comment by MarijnS95** [comment]
@kchibisov note that while searching for `VirtualKeyCode` I could still find some references in `src/platform_impl/linux/x11/events.rs` even though it was renamed to `Key` in #2662. This module doesn't seem to be referenced anywhere, was the file not deleted while X11 was refactored?

**Comment by kchibisov** [comment]
I guess everything is fine, right?

**Comment by MarijnS95** [comment]
@kchibisov it's not exactly fine as we cannot handle that keycode yet, it's not mapped.

What's worse, we don't have a way to optionally mark the keycode as used _at runtime_ (e.g. given how inconsistently we're forced to handle volume keys now since #2748) as users would likely want to back out of some menu stack. Once they are at the bottom of the stack, they will set `handled=false` on `Keycode::Back` (probably via [a `KeyEvent` extension trait](https://github.com/rust-windowing/winit/pull/2748#issuecomment-1581308061)) so that Android handles it instead and leaves or stops the app.

**Comment by kchibisov** [comment]
But you can define a filter mask on event loop builder whether we want to accept or not? Like `HashSet<Key>`. Could use something else, but setting a filter mask for events user want to handle or not sounds 'sane'.

**Comment by MarijnS95** [comment]
Was that built already? Your comment sounded like a suggestion but I'll check it out. Most important question: can we change the contents of the `HashSet` at runtime _after building the event loop_? As explained above that is vital to supporting these things.

**Comment by kchibisov** [comment]
Well, it's not here, but nothing stops you from defining such a thing. You can change the event delivery at runtime, for example we have similar concept with `listen_device_events`, you can change it at runtime just fine.

**Comment by MarijnS95** [comment]
Sure, I'll think about it but doubt I can find the time.

Am I right in assuming that you're suggesting this route because going the extension way on `KeyEvent` might be very hard to synchronize back to the platform implementation?

**Comment by kchibisov** [comment]
Right, and because events are just data which could be send between threads.

**Comment by MarijnS95** [comment]
Exactly, we have to call `finish_event(evt, handled: bool)` at some point, and have to know the value of `handled`.
